### ISSUE TITLE
TypeError: unorderable types: NoneType() < str()

### ISSUE THREAD

**Issue by PlugIN73** [docker/compose#4943]
Hi guys! I just updated to version 1.14 via pip and got error:

```
 (master) hexlet$ make compose-bash
docker-compose run web bash
Traceback (most recent call last):
  File "/usr/local/bin/docker-compose", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/main.py", line 68, in main
    command()
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/main.py", line 115, in perform_command
    project = project_from_options('.', options)
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/command.py", line 37, in project_from_options
    override_dir=options.get('--project-directory'),
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/command.py", line 91, in get_project
    config_data = config.load(config_details)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 379, in load
    service_dicts = load_services(config_details, main_file)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 475, in load_services
    service_config = merge_services(service_config, next_config)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 466, in merge_services
    for name in all_service_names
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 466, in <dictcomp>
    for name in all_service_names
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 853, in merge_service_dicts_from_files
    new_service = merge_service_dicts(base, override, version)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 929, in merge_service_dicts
    merge_ports(md, base, override)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 962, in merge_ports
    md[field] = [item for item in sorted(merged.values())]
TypeError: unorderable types: NoneType() < str()
Makefile:72: recipe for target 'compose-bash' failed
make: *** [compose-bash] Error 1
 (master) hexlet$ cat ^C
 (master) hexlet$ docker-compose --version
docker-compose version 1.14.0, build c7bdf9e
```

**Comment by shin-** [comment]
Thanks for the report, can you share your Compose file?

**Comment by PlugIN73** [comment]
Sure!
```
version: '3.1'

services:
  web-nginx:
    build:
      context: services/web_nginx
      dockerfile: Dockerfile.production
    ports:
      - "80:80"
    depends_on:
      - web

  web:
    dns:
      - 169.254.1.1
      - 8.8.8.8
    user: internal
    build:
      context: services/web
      dockerfile: Dockerfile.production
    command: bin/rails s -b 0.0.0.0
    volumes:
      - ./.irb_history:/home/internal/.irb_history
    ports:
      - "3000:3000"

  consul:
    network_mode: host
    build:
      context: services/consul
      dockerfile: Dockerfile
    environment:
      CONSUL_LOCAL_CONFIG: '{ "node_name": "exercise.hexlet.dev" }'

  registrator:
    image: gliderlabs/registrator
    network_mode: host
    command: consul://localhost:8500
    volumes:
      - '/var/run/docker.sock:/tmp/docker.sock'
    depends_on:
      - consul


  evaluator-nginx:
    build:
      context: services/evaluator_nginx
      dockerfile: Dockerfile
    environment:
      SERVICE_TAGS: evaluator
    ports:
      - "443:443"

  dnsmasq:
    network_mode: host
    build:
      context: services/dnsmasq
      dockerfile: Dockerfile
    ports:
      - "53:53/udp"
      - "53:53"
    cap_add:
      - NET_ADMIN
```

**Comment by shin-** [comment]
Hm, I can't reproduce locally. Do you have a `docker-compose.override.yml` as well, or do you provide multiple files using the `-f` option? If so, having the content of those files would be useful as well.

**Comment by PlugIN73** [comment]
docker-compose.override.yml
```
version: '3.1'

services:
  spring:
    user: internal
    build:
      context: services/web
      dockerfile: Dockerfile.development
    volumes:
      - ./services/web:/usr/src/app
    env_file: .env.development
    command: bin/spring server
    pid: host
  web-nginx:
    build:
      context: services/web_nginx
      dockerfile: Dockerfile.development
  frontend:
    build:
      context: services/web
      dockerfile: Dockerfile.development
    command: make frontend_watch
    # ports:
    #   - '8080:80'
    volumes:
      - ./services/web:/usr/src/app
  web:
    build:
      context: services/web
      dockerfile: Dockerfile.development
    env_file: .env.development
    volumes:
      - ~/.bash_history:/internal/.bash_history
      - ./services/web:/usr/src/app
    command: sh -c "rm -f tmp/pids/server.pid && bin/rails s -b 0.0.0.0"
  activejob-web:
    build:
      context: services/web
      dockerfile: Dockerfile.development
    env_file: .env.development
    volumes:
      - ~/.bash_history:/internal/.bash_history
      - ./services/web:/usr/src/app
    command: bundle exec shoryuken -R -C config/shoryuken.web.yml
  activejob-fast:
    build:
      context: services/web
      dockerfile: Dockerfile.development
    env_file: .env.development
    volumes:
      - ~/.bash_history:/internal/.bash_history
      - ./services/web:/usr/src/app
    command: bundle exec shoryuken -R -C config/shoryuken.fast.yml
  activejob-slow:
    build:
      context: services/web
      dockerfile: Dockerfile.development
    env_file: .env.development
    volumes:
      - ~/.bash_history:/internal/.bash_history
      - ./services/web:/usr/src/app
    command: bundle exec shoryuken -R -C config/shoryuken.slow.yml
```

**Comment by PlugIN73** [comment]
any docker-compose calls throws the same error... 
```
 (feature/consul-template) hexlet$ make compose-build
docker-compose build
Traceback (most recent call last):
  File "/usr/local/bin/docker-compose", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/main.py", line 68, in main
    command()
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/main.py", line 115, in perform_command
    project = project_from_options('.', options)
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/command.py", line 37, in project_from_options
    override_dir=options.get('--project-directory'),
  File "/usr/local/lib/python3.5/dist-packages/compose/cli/command.py", line 91, in get_project
    config_data = config.load(config_details)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 379, in load
    service_dicts = load_services(config_details, main_file)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 475, in load_services
    service_config = merge_services(service_config, next_config)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 466, in merge_services
    for name in all_service_names
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 466, in <dictcomp>
    for name in all_service_names
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 853, in merge_service_dicts_from_files
    new_service = merge_service_dicts(base, override, version)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 929, in merge_service_dicts
    merge_ports(md, base, override)
  File "/usr/local/lib/python3.5/dist-packages/compose/config/config.py", line 962, in merge_ports
    md[field] = [item for item in sorted(merged.values())]
TypeError: unorderable types: str() < NoneType()
Makefile:57: recipe for target 'compose-build' failed
make: *** [compose-build] Error 1
```

**Comment by shin-** [comment]
Thanks! Fix in #4955 . Workaround until then: use Python 2.7
